angband/angband
# traps: add delayed trap support and falling block traps

This change:
1) Adds a DELAY trap flag, which causes traps to trigger when the player
   leaves their square, instead of when they enter it (they can still be
   tripped by disarm attempts as normal);
2) Adds a new type of trap, a "block fall trap", which causes a granite
   wall to appear after the player leaves the grid cell the trap is on; these
   serve to block off hallways/etc behind the player as they explore.
3) Adds a new GRANITE effect that creates a granite wall at a trap's location;
4) Fixes a memory corruption bug in parse_trap_flags() that would cause it
   to try to free a non-malloced string;
5) Fixes some inconsistent indentation in load.c and save.c trap functions;
6) Removes the generic "You stumble upon a trap!" message for traps, since
   every trap has its own specific message anyway;
7) Adds a new player_leaving() hook in mon-util for effects and cleanup that
   need to happen as the player leaves a square by any means.

To test:
C-a T to create a trap, then type "block fall trap" to create one on your
grid square. Now move off it either by walking or C-a p (phase door); you
should see "A section of the ceiling falls in behind you!" and a new wall
should appear.
